acme/dns_challenge.go

```diff
@@ -19,37 +20,49 @@ var preCheckDNS preCheckDNSFunc = checkDNS
 
 var preCheckDNSFallbackCount = 5
 
-// DNSProvider represents a service for managing DNS records.
-type DNSProvider interface {
-	CreateTXTRecord(fqdn, value string, ttl int) error
-	RemoveTXTRecord(fqdn, value string, ttl int) error
+// DNSRecord returns the DNS record which will fulfill a DNS challenge
+func DNSRecord(domain, keyAuth string) (fqdn string, value string, ttl int) {
```

Comment:
This is expected to get called from Solver implementations, right? In that case we might need to document that fact on the interface.